packages/back-end/src/controllers/stripe.ts

```diff
@@ -323,3 +324,37 @@ export async function postWebhook(req: Request, res: Response) {
 
   res.status(200).send("Ok");
 }
+
+export async function getUsage(
+  req: AuthRequest<unknown, unknown, { monthsAgo?: number }>,
+  res: Response<{ status: 200; cdnUsage: DailyUsage[] }>
+) {
+  const context = getContextFromReq(req);
+
+  if (!context.permissions.canViewUsage()) {
+    context.permissions.throwPermissionError();
+  }
+
+  const monthsAgo = Math.round(req.query.monthsAgo || 0);
+  if (monthsAgo < 0 || monthsAgo > 12) {
+    throw new Error("Usage data only available for the past 12 months");
+  }
+
+  const { org } = context;
+
+  // Beginning of the month
+  const start = new Date();
+  start.setMonth(start.getMonth() - monthsAgo);
+  start.setDate(1);
+  start.setHours(0, 0, 0, 0);
+
+  // End of the month
+  const end = new Date();
+  end.setMonth(end.getMonth() - monthsAgo + 1);
+  end.setDate(0);
+  end.setHours(23, 59, 59, 999);
+
+  const cdnUsage = await getDailyCDNUsageForOrg(org.id, start, end);
+
```

Comment:
@jdorn For the short term, if we want to build this in a manner that is agnostic to the billing platform, we could calculate usage percentage here and kick off a job to fire off a sendgrid email or fire a customerio call that can kick off an email. Thoughts?

Reply:
This request is only kicked off by a user visiting the usage page.  We could use the same underlying ClickHouse data to power alerts if we want to though.